### PR TITLE
Document urlDisplayLabel to make button appear

### PR DESCRIPTION
**Pull request by neiser**
The `urlDisplayLabel` is required in order to make Grafana show a button when hovering a link over a displayed "exemplar dot". This should be documented.

See [this blog post](https://blog.qaware.de/posts/cloud-observability-grafana-spring-boot/) -> Search "Prometheus data source" for details.

### CODE REVIEW

File: docs/sources/datasources/prometheus.md [hunk 1]
```diff
@@ -286,7 +286,8 @@ datasources:
         # datasourceUid value can be anything, but it should be unique across all defined data source uids.
         - datasourceUid: my_jaeger_uid
           name: traceID
-
+          urlDisplayLabel: View
```
Comment (obetomuniz) [inline docs/sources/datasources/prometheus.md:289]: Hi @neiser, thanks for your PR, but before approving it. I want to make sure that this field is in fact, required.

On my end, I was not able to reproduce the issue indicated. It displays the `urlDisplayLabel` defined in the Data Source Settings or, by default, the Data Source name in a format like `Query with {Data Source Name}`. This rules was implemented in the PR https://github.com/grafana/grafana/pull/42732, which was introduced in Grafana 8.4.0 

Let me know if I am missing something. Also, could you please share more info about the Grafana environment (Grafana version, Operational System, etc) that you reproduced `urlDisplayLabel`  as a requirement? Thanks.

Comment (obetomuniz) [reply docs/sources/datasources/prometheus.md:289]: Trying to reproduce it with the [TNS Demo App](https://github.com/grafana/tns), it also works as expected, even without configuring the `urlDisplayLabel` field. TNS Demo App is using Grafana 8.2.2. So I assume it is not required.

https://user-images.githubusercontent.com/1680157/186511661-04155049-e5e5-413f-ad91-5edc7dcac5da.mov

Comment (neiser) [reply docs/sources/datasources/prometheus.md:289]: @obetomuniz Thanks for investigating this so diligently and sorry that I did not provide proper instructions how I encountered the error and how to reproduce it. I've encountered the error while using the kube-prometheus-stack helm chart in version 34.6.0, which uses grafana [helm chart version 6.24.1](https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-34.6.0/charts/kube-prometheus-stack/Chart.lock#L10), which in turn is using [Grafana 8.4.2]( https://github.com/grafana/helm-charts/blob/grafana-6.24.1/charts/grafana/values.yaml#L76), if I'm not completely mistaken.

I've packed all this into [this demo](https://github.com/qaware/cloud-observability-grafana-spring-boot/). Running `./run-demo.sh` spins up a local minikube and deploys Grafana stack with tempo and loki. I've just tried to reproduce it using kube-prometheus-stack version 39.2.1 and by [commenting out the crucial line](https://github.com/qaware/cloud-observability-grafana-spring-boot/blob/main/helm/kube-prometheus-stack/values.yaml#L39), but the label still appeared in my little "Spring Boot" dashboard when hovering over exemplars.

Reverting the version bump of kube-prometheus-stack still did not reproduce the issue! I think I must have encountered it with a version using grafana <8.4.0, but I can't tell how and when now (some time has indeed passed since then and creating this PR took some time).

So, long story short: You're completely right that this is fixed in #42732. Sorry that I bothered you with this.

However, wouldn't you agree that documenting this (optional) property is a good idea? Feel free to close this PR though!

Comment (obetomuniz) [reply docs/sources/datasources/prometheus.md:289]: Glad to hear it @neiser. Also, thanks for all the details about the way to reproduce it. 

Regarding the documentation, it is already [documented here](https://github.com/grafana/grafana/commit/3239c62a24a08e1ecf848f5f61db37ff356dcf5e#diff-94bbcaec653f06a96ef581b5a4f4a79cfb9d4e3792890451f4b183626a044e3cR34) as an optional value for the feature scope. Since provisioning settings are inherited from the Prometheus Settings as well, we don't need to be redundant. So I would say that we can close it.
